airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -361,7 +362,12 @@ def __init__(
         self.termination_grace_period = termination_grace_period
         self.pod_request_obj: k8s.V1Pod | None = None
         self.pod: k8s.V1Pod | None = None
-        self.skip_exit_code = skip_exit_code
+        if skip_exit_code is not None:
+            warnings.warn(
+                "skip_exit_code is deprecated. Please use skip_on_exit_code", DeprecationWarning, stacklevel=2
+            )
+            self.skip_on_exit_code = skip_exit_code
+        self.skip_on_exit_code = skip_on_exit_code
```

Comment:
Same comment here:
`self.skip_on_exit_code = skip_on_exit_code` overrides `self.skip_on_exit_code = skip_exit_code` and replaces it by `None`, you need to put it in `else`